app/models/user.rb

```diff
@@ -1,6 +1,9 @@
 class User < ApplicationRecord
   has_many :post
-  has_many :comment
+  has_many :comments, foreign_key: 'author_id'
+
+  Validate :name, presence: true
+  Validate :posts_counter, comparison: { greater_than: 0 }
```

Comment:
- [ ] in this line you need to validate if it is an integer as well by doing the following `numericality: { only_integer: true }`